### ISSUE TITLE
ungit not showing master (or origin master) and crashes when checking out branch from master

### ISSUE THREAD

**Issue by wkjid10t** [FredrikNoren/ungit#1333]
This happened recently. Master branches not appearing anymore. Literally. They don't show up in the gui. You can check them out in CLI. Then ungit still won't show them. The top right tools show that master is selected. When you attempt to check out another branch, ungit crashes. 

Screenshots of crash:
![Screenshot_20200428_010509](https://user-images.githubusercontent.com/2594108/80449191-75e7b400-88ec-11ea-8f2d-7eec231a82ec.png)
![Screenshot_20200428_010156](https://user-images.githubusercontent.com/2594108/80449192-75e7b400-88ec-11ea-91f1-f1f528900f7e.png)

**Comment by ylecuyer** [comment]
Hello could you also add the logs from the terminal where you launched
ungit. And the repo if it is not private.

**Comment by wkjid10t** [comment]
There are no logs from terminal.

**Comment by wkjid10t** [comment]
Is there a flag to output verbose/debug?

I can't share the repo. It's private.

**Comment by wkjid10t** [comment]
Master branch showed up again after manually pulling from terminal. But it only shows commits that have been made since the pull. All previous commits are not rendered. It only renders one long line... I haven't scrolled to the bottom to see where it starts.

**Comment by wkjid10t** [comment]
This is the root of my repo. The very first commit. There is something wrong with the graph. Is there cache that I can delete?

![Screenshot_20200429_182831](https://user-images.githubusercontent.com/2594108/80653110-6630b280-8a47-11ea-9d94-40b8383de5fe.png)

**Comment by jung-kim** [comment]
two settings you can set at `~/.ungitrc`. 
```
{
logGitCommands: true,
logLevel: 'info'
}
```

also, you might be at the [headless state](https://medium.com/north-arrow-research/git-tips-what-is-a-headless-state-daa9bc4e1d5a).  Can you try doing `git checkout master` first and opening up ungit?

**Comment by wkjid10t** [comment]
Not in headless state. Tried anyway, but did not fix. 

It looks like the graph is struggling with some of the merges. We had a bunch of merges back-to-back and it looks like since then, the graph just freaked out. I've tried uninstalling ungit, I've cloned the repo clean.

**Comment by wkjid10t** [comment]
This seems like a similar issue (https://github.com/FredrikNoren/ungit/issues/641)

I found that some commits that exist in the repo are not rendered but ungit. I know they aren't because I check them out by their <sha> and ungit does not render them. I can create a branch from the <sha> and ungit still does not render it. There is a commit that has a different log in ungit than in git... It looks like the ungit commit has multiple merge log messages in the commit.

**Comment by wkjid10t** [comment]
So there is a merge commit that was "missing" in ungit. I checked out the <sha> and then made a branch on that commit. The ungit rendered it. But it renders it as the "tip" of the branch, however the next commit in the chain is also rendered but they're not connected. Instead, the next commit is connected to a previous commit of the (no unmissing) "missing" commit... This seems similar to (https://github.com/FredrikNoren/ungit/issues/641), I think ungit/graph is getting confused with which commit is the parent.

**Comment by wkjid10t** [comment]
One more: I check out the "missing" commit. I make some change to source and commit. That commit now shows up as it's own standalone branch that has now root similar to the screenshot above. It still does not render the parent commit.

**Comment by jung-kim** [comment]
Right, I guess your checked out node is behind the latest node in remote or other sources.

You could set `maxActiveBranchSearchIteration` to 300 or something like that within `.ungitrc` to have ungit to go as far as n number of commits to search for checked out branch.

**Comment by wkjid10t** [comment]
That's not the issue. If you look at the earlier screenshot. I had scrolled down all the way to the very first commit. You can see how the grah "thinks" that it's not at the bottom. The reason for this is because the graph is detached somehow. What I found is when I graph log the commits with git. There are commits that exist that are NOT rendered by ungit. There are also commits rendered by ungit that contain multiple entries of merges/commits combined into 1. In other words, ungit somehow renders a commit with logs that are NOT the actual logs that are in in git. I suspect since ungit somehow made up a commit and also somehow doesn't see other commits, the graph freaks out.

**Comment by hermanus-rcaf** [comment]
I think that I may have discovered a little more information about why this issue is/was occurring. My terminal's system time was off by months (weird I know). But if I'm not mistaken, Git will take your system time and record it in the commit. 

Does ungit by any chance use the commit-time in order to display the nodes ordered by time? If so, would a commit that has out of sequence system-time cause issues?

**Comment by wkjid10t** [comment]
I think that I may have discovered a little more information about why this issue is/was occurring. My terminal's system time was off by months (weird I know). But if I'm not mistaken, Git will take your system time and record it in the commit.

Does ungit by any chance use the commit-time in order to display the nodes ordered by time? If so, would a commit that has out of sequence system-time cause issues?
